aptos-move/framework/aptos-framework/sources/vesting_without_staking.move

```diff
@@ -0,0 +1,1132 @@
+        };
+
+        vesting_schedule.last_vested_period = next_period_to_vest;
+
+        emit_event(
+            &mut vesting_contract.vest_events,
+            VestEvent {
+                admin: vesting_contract.admin,
+                vesting_contract_address: contract_address,
+                period_vested: next_period_to_vest,
+            },
+        );
+        // Every shareholder should receive the money in proportion to their shares.
+        distribute(vesting_fraction, contract_address);
+    }
+
+    /// Distribute any withdrawable grant.
+    /// This is no entry function anymore as it's called from within the vest.
+    fun distribute(vesting_fraction: FixedPoint32, contract_address: address) acquires VestingContract {
+        assert_active_vesting_contract(contract_address);
+
+        let vesting_contract = borrow_global_mut<VestingContract>(contract_address);
+        let vesting_signer = get_vesting_account_signer_internal(vesting_contract);
+        let shareholders_address = simple_map::keys(&vesting_contract.shareholders);
+        let total_amount_left = 0;
+        // Distribute coins to shareholders.
+        vector::for_each_ref(&shareholders_address, |shareholder| {
+            let shareholder = *shareholder;
+            let amount = min(simple_map::borrow(& vesting_contract.shareholders, &shareholder).left_amount,fixed_point32::multiply_u64(simple_map::borrow(& vesting_contract.shareholders, &shareholder).init_amount, vesting_fraction));
+            let recipient_address = get_beneficiary(vesting_contract, shareholder);
+            coin::transfer<AptosCoin>(&vesting_signer, recipient_address, amount);
+            let shareholder_amount = simple_map::borrow_mut(&mut vesting_contract.shareholders, &shareholder);
+            shareholder_amount.left_amount = shareholder_amount.left_amount - amount;
+            total_amount_left = total_amount_left + shareholder_amount.left_amount;
+        });
+        let total_balance = coin::balance<AptosCoin>(contract_address);
+        assert!(total_amount_left == total_balance, EBALANCE_THE_SAME);
+        if (total_balance == 0) {
+            let coins = coin::withdraw<AptosCoin>(&vesting_signer, 0);
+            coin::destroy_zero(coins);
+            set_terminate_vesting_contract(contract_address);
+        }
+    }
+
+    /// Remove the lockup period for the vesting contract. This can only be called by the admin of the vesting contract.
+    /// Example usage: If admin find shareholder suspicious, admin can remove it.
+    public entry fun remove_shareholder(admin: &signer, contract_address: address, shareholder_address: address) acquires VestingContract {
+        let vesting_contract = borrow_global_mut<VestingContract>(contract_address);
+        verify_admin(admin, vesting_contract);
+        let shareholders = &mut vesting_contract.shareholders;
+        simple_map::remove(shareholders, &shareholder_address);
```

Comment:
Just a remark, when we `remove`, and do not transfer the balance in `withdraw_address`, there would be imbalance in the actual coins with `vesting_contract` and sum of `shareholder.left_amount`. This would result in assertion failure in `distribute`. Perhaps we should change the assertion to `total_mount_left <= balance_in_contract` and terminate the contract when `total_amount_left` becomes zero.

There should also be `ShareHolderRemovedEvent` with fields that provides `shareholder, beneficiary, amount`.

Reply:
@axiongsupra , yes perhaps that is a good idea. When `shareholder` is removed, withdraw the `left_amount` to `withdraw_address`. Appropriate withdraw event should be emitted in that case.

Reply:
Okay, sounds good to me.